python/az/aro/azext_aro/custom.py

```diff
@@ -211,65 +211,28 @@ def aro_list_credentials(client, resource_group_name, resource_name):
     return client.list_credentials(resource_group_name, resource_name)
 
 
-def aro_update(cmd, client, resource_group_name, resource_name, no_wait=False):
-    rp_client_sp = None
-    client_sp = None
-    resources = set()
-
-    try:
-        oc = client.get(resource_group_name, resource_name)
-
-        # Get cluster resources we need to assign network contributor on
-        resources = get_cluster_network_resources(cmd.cli_ctx, oc)
-    except (CloudError, HttpOperationError) as e:
-        logger.info(e.message)
-
-    aad = AADManager(cmd.cli_ctx)
-
-    if rp_mode_production():
-        rp_client_id = FP_CLIENT_ID
-    else:
-        rp_client_id = os.environ.get('AZURE_FP_CLIENT_ID', FP_CLIENT_ID)
-
-    # Best effort - assume the role assignments on the SP exist if exception raised
-    try:
-        rp_client_sp = aad.get_service_principal(rp_client_id)
-        if not rp_client_sp:
-            raise ResourceNotFoundError("RP service principal not found.")
-    except GraphErrorException as e:
-        logger.info(e.message)
-
-    client_id = oc.service_principal_profile.client_id
+def aro_update(cmd,
+               client,
+               resource_group_name,
+               resource_name,
+               client_id=None,
+               client_secret=None,
+               no_wait=False):
+    # if we can't read cluster spec, we will not be able to do much. Fail.
+    oc = client.get(resource_group_name, resource_name)
 
-    # Best effort - assume the role assignments on the SP exist if exception raised
-    try:
-        client_sp = aad.get_service_principal(client_id)
-        if not client_sp:
-            raise ResourceNotFoundError("Cluster service principal not found.")
-    except GraphErrorException as e:
-        logger.info(e.message)
+    client_id, client_secret = service_principal_update(cmd.cli_ctx, oc, client_id, client_secret)
 
-    # Drop any None service principal objects
-    sp_obj_ids = [sp.object_id for sp in [rp_client_sp, client_sp] if sp]
+    # construct update payload
+    oc = openshiftcluster.OpenShiftClusterUpdate(
+        service_principal_profile=openshiftcluster.ServicePrincipalProfile(),
```

Comment:
this shouldn't be added if the cluster SP isn't being touched, right?

Reply:
yes